src/analysis/convex/strict_convex_space.lean

```diff
@@ -238,3 +238,149 @@ lemma norm_midpoint_lt_iff (h : ∥x∥ = ∥y∥) : ∥(1/2 : ℝ) • (x + y)
+maps 0 to 0.  Unlike Mazur-Ulam, this does not require the isometry to be surjective.  -/
+def linear_isometry_of_map_zero {f : F → E} (hi : isometry f) (h0 : f 0 = 0) :
+  F →ₗᵢ[ℝ] E :=
+{ to_fun := f,
+  map_add' := hi.map_add_of_map_zero h0,
+  map_smul' := hi.map_smul_of_map_zero h0,
+  norm_map' := hi.norm_map_of_map_zero h0 }
+
+@[simp] lemma coe_linear_isometry_of_map_zero {f : F → E} (hi : isometry f)
+  (h0 : f 0 = 0) :
+  ⇑(hi.linear_isometry_of_map_zero h0) = f :=
+rfl
+
+variables {PF : Type*} {PE : Type*} [metric_space PF] [metric_space PE]
+variables [normed_add_torsor F PF] [normed_add_torsor E PE]
+include E F
+
+/-- An isometry of `normed_add_torsor`s for real normed spaces, strictly convex in the case of
+the codomain, induces a linear isometry at any point.  Unlike Mazur-Ulam, this does not require
+the isometry to be surjective.  -/
+def linear_isometry_at {f : PF → PE} (hi : isometry f) (p : PF) : F →ₗᵢ[ℝ] E :=
+linear_isometry_of_map_zero
+  (show isometry (λ x : F, f (x +ᵥ p) -ᵥ f p), begin
+    intros x y,
+    simp_rw [edist_dist, dist_vsub_cancel_right, hi.dist_eq, dist_eq_norm_vsub,
+             vadd_vsub_vadd_cancel_right, vsub_eq_sub]
+   end) (by simp)
+
+@[simp] lemma coe_linear_isometry_at {f : PF → PE} (hi : isometry f) (p : PF) :
+  ⇑(hi.linear_isometry_at p) = λ x : F, f (x +ᵥ p) -ᵥ f p :=
+rfl
+
+lemma linear_isometry_at_apply {f : PF → PE} (hi : isometry f) (p : PF) (v : F) :
+  hi.linear_isometry_at p v = f (v +ᵥ p) -ᵥ f p :=
+rfl
+
+lemma linear_isometry_at_apply_vsub {f : PF → PE} (hi : isometry f)
+  (p₁ p₂ : PF) :
+  hi.linear_isometry_at p₁ (p₂ -ᵥ p₁) +ᵥ f p₁ = f p₂ :=
+by simp
+
+lemma linear_isometry_at_eq {f : PF → PE} (hi : isometry f) (p₁ p₂ : PF) :
+  hi.linear_isometry_at p₁ = hi.linear_isometry_at p₂ :=
+begin
+  ext x,
+  rw [hi.linear_isometry_at_apply, hi.linear_isometry_at_apply,
+      ←hi.linear_isometry_at_apply_vsub p₁ (x +ᵥ p₂), ←hi.linear_isometry_at_apply_vsub p₁ p₂,
+      vadd_vsub_vadd_cancel_right, ←linear_isometry.map_sub, hi.linear_isometry_at_apply,
+      vsub_sub_vsub_cancel_right, vadd_vsub]
+end
+
+/-- An isometry of `normed_add_torsor`s for real normed spaces, strictly convex in the case of
+the codomain, is an affine isometry.  Unlike Mazur-Ulam, this does not require the isometry to
+be surjective.  -/
+noncomputable def affine_isometry_of_strict_convex_space {f : PF → PE} (hi : isometry f) :
+  PF →ᵃⁱ[ℝ] PE :=
+{ to_fun := f,
+  linear := (hi.linear_isometry_at (classical.arbitrary PF)).to_linear_map,
+  map_vadd' := λ p v, begin
+    rw hi.linear_isometry_at_eq (classical.arbitrary PF) p,
+    simp
+  end,
+  norm_map := (hi.linear_isometry_at _).norm_map }
+
+@[simp] lemma coe_affine_isometry_of_strict_convex_space {f : PF → PE} (hi : isometry f) :
+  ⇑(hi.affine_isometry_of_strict_convex_space) = f :=
+rfl
+
+lemma affine_isometry_of_strict_convex_space_apply {f : PF → PE} (hi : isometry f)
+  (p : PF) :
+  hi.affine_isometry_of_strict_convex_space p = f p :=
+rfl
```

Comment:
This too can be simp, because it's an applied lemma.

Reply:
I've marked this as simp, is there documentation somewhere of when it's appropriate to mark a lemma as simp when it can be proved by simp, and when that will make the simp linter fail?

Reply:
This is the only case I'm aware of, namely pairs of lemmas `⇑f = g` and `⇑f a = g a`.